### PR TITLE
Implement User Data Fetch and Error Handling in UserContext

### PR DESCRIPTION
**Pull request by lugenx**
This pull request addresses issue #209 

Changes made:

- UserContext now fetches user data based on the token and stores it in the state.
- Added error handling during the fetch process. If an error occurs, it will be stored in the error state.
- The user profile component now displays the fetched user data.
- UserContext now provides the user data, token, and any potential errors to all components in the application.

Please review the changes and provide your feedback.

### CODE REVIEW

File: client/src/pages/ProfilePage.js [hunk 1]
```diff
@@ -17,15 +18,16 @@ import {
...
 const ProfilePage = () => {
   // TEMPORARY: PLACEHOLDER DATA
-  const [user, setUser] = useState({
-    name: "Ralph Macchio",
-    email: "thekaratekid@gmail.com",
-    createdAt: "02/24/2023",
-    postalCode: "02116",
-    roles: [],
-  });
+  // const [user, setUser] = useState({
```
Comment (tsimian) [inline client/src/pages/ProfilePage.js:21]: Perhaps we can remove the placeholder data now, what do you think?

**Review by tsimian — APPROVED** [review]
I just left a single comment, but this looks good to me @lugenx. Nice work!

**Comment by lugenx** [comment]
i addressed your comment @tsimian. Thank you.

**Review by tsimian — APPROVED** [review]
Looks good to me @lugenx

**Pull request merged**